payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/PostalCodeConfig.kt

```diff
@@ -0,0 +1,81 @@
+package com.stripe.android.ui.core.elements
+
+import androidx.annotation.StringRes
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.VisualTransformation
+import kotlinx.coroutines.flow.MutableStateFlow
+
+internal class PostalCodeConfig(
+    @StringRes override val label: Int,
+    override val capitalization: KeyboardCapitalization = KeyboardCapitalization.Words,
+    override val keyboard: KeyboardType = KeyboardType.Text,
+    override val trailingIcon: MutableStateFlow<TextFieldIcon?> = MutableStateFlow(null),
+    country: String
+) : TextFieldConfig {
+    override val debugLabel: String = "postal_code_text"
+    override val visualTransformation: VisualTransformation? = null
+    override val loading: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    private val format = CountryPostalFormat.forCountry(country)
+
+    override fun determineState(input: String): TextFieldState = object : TextFieldState {
+        override fun shouldShowError(hasFocus: Boolean) = false
+
+        override fun isValid(): Boolean {
+            val formatted = input.replace(Regex("\\s+"), "")
+            return formatted.length in format.minimumLength..format.maximumLength &&
+                input.matches(format.regexPattern)
+        }
+
+        override fun getError(): FieldError? = null
+
+        override fun isFull(): Boolean = input.length >= format.minimumLength
+
+        override fun isBlank(): Boolean = input.isBlank()
+    }
+
+    override fun filter(userTyped: String): String =
+        if (
+            setOf(KeyboardType.Number, KeyboardType.NumberPassword).contains(keyboard)
+        ) {
+            userTyped.filter { it.isDigit() }
+        } else {
+            userTyped
+        }
+
+    override fun convertToRaw(displayName: String) = displayName
+
+    override fun convertFromRaw(rawValue: String) = rawValue
+
+    sealed class CountryPostalFormat(
+        val minimumLength: Int,
+        val maximumLength: Int,
+        val regexPattern: Regex
+    ) {
+        object CA : CountryPostalFormat(
+            minimumLength = 6,
+            maximumLength = 6,
+            regexPattern = Regex("[a-zA-Z]\\d[a-zA-Z][\\s-]?\\d[a-zA-Z]\\d")
+        )
+        object US : CountryPostalFormat(
+            minimumLength = 5,
+            maximumLength = 9,
+            regexPattern = Regex("\\d+")
+        )
+        object Other : CountryPostalFormat(
```

Comment:
Is this copied from stripe.js or somewhere else? Just making sure we're not being too strict on the requirements. Better to not validate than to validate it wrong and end up not accepting a valid value, which frustrates the user.